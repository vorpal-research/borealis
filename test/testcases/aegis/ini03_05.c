int main(void) {
    int **p;
    
    int *q = *p;
    int i = *q;    
    
    return i;
}
