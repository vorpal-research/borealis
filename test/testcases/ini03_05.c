//
// Разыменование неинициализированного или нулевого указателя
//

int main(void) {
    int **p;
    
    int *q = *p;   // дефект INI-03, отрубили ветку
    int i = *q;    
    
    return i;
}
