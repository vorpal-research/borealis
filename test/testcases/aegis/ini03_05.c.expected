
[
	
	{
		"location" : 
		{
			"filename" : "test/testcases/aegis/ini03_05.c",
			"loc" : 
			{
				"col" : 16,
				"line" : 4
			}
		},
		"type" : "NDF-01"
	},
	
	{
		"location" : 
		{
			"filename" : "test/testcases/aegis/ini03_05.c",
			"loc" : 
			{
				"col" : 15,
				"line" : 5
			}
		},
		"type" : "NDF-01"
	},
	
	{
		"location" : 
		{
			"filename" : "test/testcases/aegis/ini03_05.c",
			"loc" : 
			{
				"col" : 5,
				"line" : 7
			}
		},
		"type" : "NDF-01"
	}
]
